src/test/ui/impl-trait/must_outlive_least_region_or_bound.nll.stderr

```diff
@@ -41,12 +41,20 @@ error: unsatisfied lifetime constraints
    |
 LL | fn explicit<'a>(x: &'a i32) -> impl Copy { x }
    |             -- lifetime `'a` defined here  ^ returning this value requires that `'a` must outlive `'static`
+help: you can add a constraint to the return type to make it last less than `'static` and match 'a
+   |
+LL | fn explicit<'a>(x: &'a i32) -> impl Copy + 'a { x }
+   |                                ^^^^^^^^^^^^^^
 
 error: unsatisfied lifetime constraints
   --> $DIR/must_outlive_least_region_or_bound.rs:22:69
    |
 LL | fn with_bound<'a>(x: &'a i32) -> impl LifetimeTrait<'a> + 'static { x }
    |               -- lifetime `'a` defined here                         ^ returning this value requires that `'a` must outlive `'static`
+help: you can add a constraint to the return type to make it last less than `'static` and match 'a
+   |
+LL | fn with_bound<'a>(x: &'a i32) -> impl LifetimeTrait<'a> + 'static + 'a { x }
+   |                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
Not sure how problematic the suggestion is in this case where there's already a bound.

Reply:
The correct suggestion then would probably be to either _replace_ the existing bound or to change the signature to `fn with_bound<'a: 'static>`. For the latter it might be possible to get the `DefId` of `'a`, which would let you get the correct span. Looking at the signature, for anyone with some experience will notice that `'a` _has_ to be `'a: 'static`, but for it'd be nice to also have a `note` for this case explaining the case.

If catching the case is too hard, I'd be ok with merging as is and filing an issue for it.

Reply:
I've spent a little while trying to implement your latter suggestion and I can't quite figure out how to do it. I'd like to see it in this PR though.

Firstly, I'm not sure how to check for the `+ 'static` part of the return type - I looked in the return type `TyKind::Anon(_, substs)` and in the function `TyKind::FnDef(_, substs)` and then in output from `tcx.fn_sig(..)` for the type and couldn't find anything related to that bound. If you could point me in the direction of where I might be able to observe that bound, that'd be helpful (cc @nikomatsakis).

Secondly, after putting a ugly fix in place to get around the above issue temporarily, I started trying to get the `Span` for the `'a` in `fn_sig<'a>(..)`. To do that I took the `Region` that I had that would give me `"'a"` when I worked out the name and called `free_region_binding_scope(..)` on it to get a `DefId`, then calling `tcx.def_span` to get a `Span` from that. Sadly that gave a span for the whole line (or at least the end of it, because that's where the addition from the `format!(..)` call went). So I didn't really make much progress on that half of the suggestion either before calling it in for the night.. any ideas here might be appreciated too.

Reply:
Fixed this now.

Reply:
Can you double check the wording on this new suggestion @estebank?